### PR TITLE
Implement continue

### PR DESCRIPTION
**Pull request by anukul**
This change will be present in the next Storyscript compiler release.

### CODE REVIEW

File: asyncy/processing/Lexicon.py [hunk 1]
```diff
@@ -130,6 +130,16 @@ def _does_line_have_parent_method(story, line, parent_method_wanted):
             # There is no parent, this is an illegal usage of break.
             raise InvalidKeywordUsage(story, line, 'break')
 
+    @staticmethod
+    async def continue_(logger, story, line):
+        # Ensure that we're in a foreach loop. If we are, return CONTINUE,
+        # otherwise raise an exception.
+        if Lexicon._does_line_have_parent_method(story, line, 'for'):
```
Comment (wilzbach) [inline asyncy/processing/Lexicon.py]: What about `while`?

File: tests/integration/processing/Lexicon.py [hunk 2]
```diff
@@ -261,6 +261,17 @@ def __init__(self, cases, preparation_lines=''):
...
     ),
+    TestSuite(
+        preparation_lines='a = [1, 1, 1, 2, 3, 4, 5]\n'
+                          'b = 0\n',
+        cases=[
+            TestCase(append='foreach a as elem\n'
+                            '   if elem % 2 == 0\n'
+                            '       continue\n'
+                            '   b = b + elem\n',
+                     assertion=ContextAssertion(key='b', expected=11))
```
Comment (wilzbach) [inline tests/integration/processing/Lexicon.py]: Needs a test case for `while` too.

**Comment by anukul** [comment]
@seb There is no `while` yet in the engine.
I created a separate issue, and will probably get to it tomorrow.

**Comment by judepereira** [comment]
@wilzbach can we lower foreach with while? then the engine no longer needs to support both while and foreach

**Comment by wilzbach** [comment]
> @wilzbach can we lower foreach with while? then the engine no longer needs to support both while and foreach

Yes, I thought about this a while ago too. See e.g. https://github.com/storyscript/storyscript/issues/703

**Comment by anukul** [comment]
@judepereira @wilzbach ready for review

File: tests/integration/processing/Lexicon.py (hunk 2 above)
Comment (judepereira) [reply tests/integration/processing/Lexicon.py]: @anukul test for while?

Comment (anukul) [reply tests/integration/processing/Lexicon.py]: There is no `while` in the engine yet.

Comment (judepereira) [reply tests/integration/processing/Lexicon.py]: Oh, okay then.

**Pull request merged**